{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Tracking \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Initializing your tracker (PyPads)\n",
    "First you have to install pypads-padre which has pypads as a dependency\n",
    "\n",
    "    pip install pypads-padre"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Trying to configure padre plugin for pypads...\n"
     ]
    }
   ],
   "source": [
    "from pypads.app.base import PyPads\n",
    "tracker = PyPads(autostart=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3D MNIST example"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-11-12 11:39:17.437 | INFO     | pypads.app.base:activate_tracking:566 - Activating tracking by extending importlib...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/ipykernel/ipkernel.py:287: DeprecationWarning: `should_run_async` will not call `transform_cell` automatically in the future. Please pass the result to `transformed_cell` argument and any exception that happen during thetransform in `preprocessing_exc_tuple` in IPython 7.17 and above.\n",
      "  and should_run_async(code)\n",
      "Exception in thread Thread-8:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 349, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-9:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 349, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-10:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 349, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-11:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 349, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-13:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 362, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-14:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 349, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-15:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-11-12 11:39:18.940 | INFO     | pypads.injections.setup.misc_setup:_call:74 - Tracking execution to run with id c35525d5359b460ba03ec75ef2826d61\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-16:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-17:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 362, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-18:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-19:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-20:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-21:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-22:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-23:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-24:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-25:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-26:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-27:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-29:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-30:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-31:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-32:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-33:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-34:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Exception in thread Thread-35:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-36:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-37:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-38:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-11-12 11:39:21.691 | WARNING  | pypads.app.api:track:127 - Tracking a function without a mapping definition. A default mapping will be generated.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-39:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-40:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-41:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-42:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-43:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-44:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-45:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 362, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-46:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-47:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-48:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-49:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-50:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-51:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-11-12 11:39:28.612 | WARNING  | pypads.app.api:track:127 - Tracking a function without a mapping definition. A default mapping will be generated.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-52:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-53:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-11-12 11:39:30.183 | INFO     | pypads.injections.loggers.debug:__pre__:27 - Pypads tracked class <class 'sklearn.linear_model.logistic.LogisticRegression'> initialized.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-54:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-55:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-11-12 11:39:30.827 | WARNING  | pypads.injections.analysis.parameters:__post__:128 - No parameters are defined on the mapping file for <class 'sklearn.preprocessing.label.LabelEncoder'>. Trying to log parameters without schema definition programmatically.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.\n",
      "  FutureWarning)\n",
      "/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/sklearn/linear_model/logistic.py:469: FutureWarning: Default multi_class will be changed to 'auto' in 0.22. Specify the multi_class option to silence this warning.\n",
      "  \"this warning.\", FutureWarning)\n",
      "Exception in thread Thread-56:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-57:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n",
      "Exception in thread Thread-58:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 926, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/pypads_onto/injections/converter.py\", line 291, in parse\n",
      "    graph.parse(data=entry.json(by_alias=True), format=\"json-ld\")\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 1078, in parse\n",
      "    parser.parse(source, self, **args)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 95, in parse\n",
      "    to_rdf(data, conj_sink, base, context_data)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 107, in to_rdf\n",
      "    return parser.parse(data, context, dataset)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 140, in parse\n",
      "    self._add_to_graph(dataset, graph, context, node, topcontext)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 177, in _add_to_graph\n",
      "    no_id=no_id)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib_jsonld/parser.py\", line 262, in _key_to_graph\n",
      "    graph.add((subj, pred, obj))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/graph.py\", line 389, in add\n",
      "    self.__store.add((s, p, o), self, quoted=False)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 551, in add\n",
      "    self.commit()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 527, in commit\n",
      "    self._update('\\n;\\n'.join(self._edits))\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlstore.py\", line 606, in _update\n",
      "    SPARQLConnector.update(self, update)\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/rdflib/plugins/stores/sparqlconnector.py\", line 124, in update\n",
      "    res.raise_for_status()\n",
      "  File \"/home/mehdi/anaconda3/envs/pypads-padre-3.7/lib/python3.7/site-packages/requests/models.py\", line 941, in raise_for_status\n",
      "    raise HTTPError(http_error_msg, response=self)\n",
      "requests.exceptions.HTTPError: 503 Server Error: Service Unavailable for url: http://rdf.padre-lab.eu/pypads/update\n",
      "\n"
     ]
    }
   ],
   "source": [
    "tracker.start_track(experiment_name= \"3D MNIST\")\n",
    "import h5py\n",
    "import numpy as np\n",
    "\n",
    "\n",
    "# Loading and tracking your dataset\n",
    "path = \"data/3d-mnist/full_dataset_vectors.h5\"\n",
    "@tracker.decorators.dataset(name=\"3DMNIST\", target_columns=[-1])\n",
    "def load_3d_mnist(path):\n",
    "    \"\"\"\n",
    "    The aim of this dataset is to provide a simple way to get started with 3D computer vision problems such as 3D shape recognition.\n",
    "\n",
    "    Accurate 3D point clouds can (easily and cheaply) be adquired nowdays from different sources:\n",
    "\n",
    "     - RGB-D devices: Google Tango, Microsoft Kinect, etc.\n",
    "     - Lidar.\n",
    "     - 3D reconstruction from multiple images.\n",
    "\n",
    "    However there is a lack of large 3D datasets (you can find a good one here based on triangular meshes); it's especially hard to find datasets based on point clouds (wich is the raw output from every 3D sensing device).\n",
    "\n",
    "    This dataset contains 3D point clouds generated from the original images of the MNIST dataset to bring a familiar introduction to 3D to people used to work with 2D datasets (images).\n",
    "\n",
    "    The full dataset is splitted into arrays:\n",
    "\n",
    "    X_train (10000, 4096)\n",
    "    y_train (10000)\n",
    "    X_test(2000, 4096)\n",
    "    y_test (2000)\n",
    "    \n",
    "    data is the concatenation of X_train, X_test, y_train, y_test (12000, 4097)\n",
    "    \n",
    "    :return data \n",
    "    \"\"\"\n",
    "    with h5py.File(path, \"r\") as hf:\n",
    "        X_train, y_train = hf[\"X_train\"][:], hf[\"y_train\"][:]\n",
    "        X_test, y_test = hf[\"X_test\"][:], hf[\"y_test\"][:]\n",
    "        train_data = np.concatenate([X_train, y_train.reshape(len(y_train), 1)], axis=1)\n",
    "        test_data = np.concatenate([X_test, y_test.reshape(len(y_test), 1)], axis=1)\n",
    "        data = np.concatenate([train_data, test_data], axis=0)\n",
    "    return data\n",
    "\n",
    "\n",
    "data = load_3d_mnist(path)\n",
    "\n",
    "# Tracking your train and test splits: in this simple case we follow the original data train : (10000) test: (2000)\n",
    "\n",
    "@tracker.decorators.splitter()\n",
    "def splitter(data, index=10000):\n",
    "    import numpy as np\n",
    "    idx = np.arange(data.shape[0])\n",
    "    return idx[:index], idx[index:]\n",
    "\n",
    "train, test = splitter(data)\n",
    "# Do and log stuff\n",
    "from sklearn.linear_model import LogisticRegression\n",
    "from sklearn.metrics.classification import f1_score\n",
    "\n",
    "X_train, y_train = data[train,:-1],data[train,-1]\n",
    "X_test, y_test = data[test,:-1],data[test,-1]\n",
    "\n",
    "model = LogisticRegression()\n",
    "\n",
    "model.fit(X_train, y_train)\n",
    "\n",
    "preds = model.predict(X_test)\n",
    "\n",
    "f1 = f1_score(preds,y_test, average=\"macro\")\n",
    "\n",
    "print(\"F1_score: \", str(f1))\n",
    "\n",
    "# end_run\n",
    "tracker.api.end_run()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Pytorch example of 3d Mnist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Activate tracking of pypads\n",
    "tracker = PyPads(autostart=\"3D-MNIST-Torch\", setup_fns=[])\n",
    "\n",
    "import torch\n",
    "from torch import nn\n",
    "from torch.autograd import Variable\n",
    "import h5py\n",
    "import numpy as np\n",
    "\n",
    "@tracker.decorators.dataset(name=\"3D-MNIST\", target_columns=[-1])\n",
    "def load_3d_mnist(path):\n",
    "    \"\"\"\n",
    "    The aim of this dataset is to provide a simple way to get started with 3D computer vision problems such as 3D shape recognition.\n",
    "\n",
    "    Accurate 3D point clouds can (easily and cheaply) be adquired nowdays from different sources:\n",
    "\n",
    "     - RGB-D devices: Google Tango, Microsoft Kinect, etc.\n",
    "     - Lidar.\n",
    "     - 3D reconstruction from multiple images.\n",
    "\n",
    "    However there is a lack of large 3D datasets (you can find a good one here based on triangular meshes); it's especially hard to find datasets based on point clouds (wich is the raw output from every 3D sensing device).\n",
    "\n",
    "    This dataset contains 3D point clouds generated from the original images of the MNIST dataset to bring a familiar introduction to 3D to people used to work with 2D datasets (images).\n",
    "\n",
    "    The full dataset is splitted into arrays:\n",
    "\n",
    "    X_train (10000, 4096)\n",
    "    y_train (10000)\n",
    "    X_test(2000, 4096)\n",
    "    y_test (2000)\n",
    "\n",
    "    \"\"\"\n",
    "\n",
    "    with h5py.File(path, \"r\") as hf:\n",
    "        X_train, y_train = hf[\"X_train\"][:], hf[\"y_train\"][:]\n",
    "        X_test, y_test = hf[\"X_test\"][:], hf[\"y_test\"][:]\n",
    "        train_data = np.concatenate([X_train, y_train.reshape(len(y_train), 1)], axis=1)\n",
    "        test_data = np.concatenate([X_test, y_test.reshape(len(y_test), 1)], axis=1)\n",
    "        data = np.concatenate([train_data, test_data], axis=0)\n",
    "    return data\n",
    "\n",
    "class CNNModel(nn.Module):\n",
    "    def __init__(self, dim_output):\n",
    "        super(CNNModel, self).__init__()\n",
    "\n",
    "        self.conv_layer1 = self._conv_layer_set(3, 32)\n",
    "        self.conv_layer2 = self._conv_layer_set(32, 64)\n",
    "        self.fc1 = nn.Linear(2 ** 3 * 64, 128)\n",
    "        self.fc2 = nn.Linear(128, dim_output)\n",
    "        self.relu = nn.LeakyReLU()\n",
    "        self.batch = nn.BatchNorm1d(128)\n",
    "        self.drop = nn.Dropout(p=0.15)\n",
    "\n",
    "    def _conv_layer_set(self, in_c, out_c):\n",
    "        conv_layer = nn.Sequential(\n",
    "            nn.Conv3d(in_c, out_c, kernel_size=(3, 3, 3), padding=0),\n",
    "            nn.LeakyReLU(),\n",
    "            nn.MaxPool3d((2, 2, 2)),\n",
    "        )\n",
    "        return conv_layer\n",
    "\n",
    "    @tracker.decorators.track(event=['pypads_predict'])\n",
    "    def forward(self, x):\n",
    "        # Set 1\n",
    "        out = self.conv_layer1(x)\n",
    "        out = self.conv_layer2(out)\n",
    "        out = out.view(out.size(0), -1)\n",
    "        out = self.fc1(out)\n",
    "        out = self.relu(out)\n",
    "        out = self.batch(out)\n",
    "        out = self.drop(out)\n",
    "        out = self.fc2(out)\n",
    "\n",
    "        return out\n",
    "\n",
    "# Sample shape\n",
    "sample_shape = (16, 16, 16, 3)\n",
    "\n",
    "# Load 3d Mnist data\n",
    "path = \"/home/mehdi/Desktop/research_assistant_236/PyPadre/pypads-examples/Notebooks-DataScience Lab/\" \\\n",
    "       \"data/3d-mnist/full_dataset_vectors.h5\"\n",
    "data = load_3d_mnist(path)\n",
    "X_train, y_train = data[:100, :-1], data[:100, -1]\n",
    "X_test, y_test = data[11900:, :-1], data[11900:, -1]\n",
    "\n",
    "# Reshape data into 3D format (16,16,16)\n",
    "X_train = data_transform(X_train, sample_shape)\n",
    "X_test = data_transform(X_test, sample_shape)\n",
    "\n",
    "train_x = torch.from_numpy(X_train).float()\n",
    "train_y = torch.from_numpy(y_train).long()\n",
    "test_x = torch.from_numpy(X_test).float()\n",
    "test_y = torch.from_numpy(y_test).long()\n",
    "\n",
    "# Pytorch train and test sets\n",
    "train = torch.utils.data.TensorDataset(train_x, train_y)\n",
    "test = torch.utils.data.TensorDataset(test_x, test_y)\n",
    "\n",
    "# Definition of hyperparameters\n",
    "batch_size = 10\n",
    "tracker.api.log_param('batch_size', batch_size, description=\"Batch size\")\n",
    "num_epochs = 10\n",
    "tracker.api.log_param('num_epochs', num_epochs, description=\"Number of training epochs\")\n",
    "\n",
    "# Data Loader\n",
    "train_loader = torch.utils.data.DataLoader(train, batch_size=batch_size, shuffle=False)\n",
    "test_loader = torch.utils.data.DataLoader(test, batch_size=batch_size, shuffle=False)\n",
    "\n",
    "# Create CNN\n",
    "model = CNNModel(10)\n",
    "# model.cuda()\n",
    "print(model)\n",
    "\n",
    "# Cross Entropy Loss\n",
    "error = nn.CrossEntropyLoss()\n",
    "\n",
    "# SGD Optimizer\n",
    "learning_rate = 0.001\n",
    "tracker.api.log_param('learning_rate', learning_rate, description=\"Learning rate of the optimizer\")\n",
    "optimizer = torch.optim.SGD(model.parameters(), lr=learning_rate)\n",
    "\n",
    "# CNN model training\n",
    "count = 0\n",
    "loss_list = []\n",
    "iteration_list = []\n",
    "accuracy_list = []\n",
    "for epoch in range(num_epochs):\n",
    "    for i, (images, labels) in enumerate(train_loader):\n",
    "\n",
    "        train = Variable(images.view(100, 3, 16, 16, 16))\n",
    "        labels = Variable(labels)\n",
    "        # Clear gradients\n",
    "        optimizer.zero_grad()\n",
    "        # Forward propagation\n",
    "        outputs = model(train)\n",
    "        # Calculate softmax and ross entropy loss\n",
    "        loss = error(outputs, labels)\n",
    "        # Calculating gradients\n",
    "        loss.backward()\n",
    "        # Update parameters\n",
    "        optimizer.step()\n",
    "\n",
    "        count += 1\n",
    "        if count % 50 == 0:\n",
    "            # Calculate Accuracy\n",
    "            correct = 0\n",
    "            total = 0\n",
    "            # Iterate through test dataset\n",
    "            for images, labels in test_loader:\n",
    "                test = Variable(images.view(100, 1, 16, 16, 16))\n",
    "                # Forward propagation\n",
    "                outputs = model(test)\n",
    "\n",
    "                # Get predictions from the maximum value\n",
    "                predicted = torch.max(outputs.data, 1)[1]\n",
    "\n",
    "                # Total number of labels\n",
    "                total += len(labels)\n",
    "                correct += (predicted == labels).sum()\n",
    "\n",
    "            accuracy = 100 * correct / float(total)\n",
    "\n",
    "            # store loss and iteration\n",
    "            loss_list.append(loss.data)\n",
    "            iteration_list.append(count)\n",
    "            accuracy_list.append(accuracy)\n",
    "\n",
    "            if count % 500 == 0:\n",
    "                # Print Loss\n",
    "                print('Iteration: {}  Loss: {}  Accuracy: {} %'.format(count, loss.data, accuracy))\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Covid-19 tweets example"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tracker.start_track(experiment_name= \"Covid-19 Tweets\")\n",
    "import pandas as pd\n",
    "from pandas_profiling import ProfileReport\n",
    "\n",
    "# Loading and tracking your dataset\n",
    "path = \"data/covid-19-tweets/covid19_tweets.csv\"\n",
    "@tracker.decorators.dataset(name=\"Covid-19 Tweets\")\n",
    "def load_covid_tweets(path):\n",
    "    \"\"\"\n",
    "    These tweets are collected using Twitter API and a Python script. \n",
    "    A query for this high-frequency hashtag (#covid19) is run on a daily basis for a certain time period, to collect a larger number of tweets samples.\n",
    "    The collection script can be found here: https://github.com/gabrielpreda/covid-19-tweets\n",
    "    \n",
    "    :return dataframe \n",
    "    \"\"\"\n",
    "    df = pd.read_csv(path)\n",
    "    return df\n",
    "\n",
    "\n",
    "df = load_covid_tweets(path)\n",
    "\n",
    "# Do and log stuff\n",
    "profile = ProfileReport(df, title=\"Covid tweets\")\n",
    "\n",
    "# end_run\n",
    "tracker.api.end_run()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "profile"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Corona Virus 2019 Dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "tracker.start_track(experiment_name = \"Corona Virus 2019 Dataset\")\n",
    "import pandas as pd\n",
    "from pandas_profiling import ProfileReport\n",
    "\n",
    "# Loading \n",
    "path = \"data/covid-19/covid_19_data.csv\"\n",
    "\n",
    "@tracker.decorators.dataset(name=\"Covid-19 cases Dataset\")\n",
    "def load_covid_dataset(path):\n",
    "    \"\"\"\n",
    "    Content\n",
    "\n",
    "    2019 Novel Coronavirus (2019-nCoV) is a virus (more specifically, a coronavirus) identified as the cause of an outbreak of respiratory illness first detected in Wuhan, China. Early on, many of the patients in the outbreak in Wuhan, China reportedly had some link to a large seafood and animal market, suggesting animal-to-person spread. However, a growing number of patients reportedly have not had exposure to animal markets, indicating person-to-person spread is occurring. At this time, it’s unclear how easily or sustainably this virus is spreading between people - CDC\n",
    "\n",
    "    This dataset has daily level information on the number of affected cases, deaths and recovery from 2019 novel coronavirus. Please note that this is a time series data and so the number of cases on any given day is the cumulative number.\n",
    "\n",
    "    The data is available from 22 Jan, 2020.\n",
    "    Column Description\n",
    "\n",
    "    Main file in this dataset is covid_19_data.csv and the detailed descriptions of the features are below.\n",
    "\n",
    "        - Sno - Serial number\n",
    "        - ObservationDate - Date of the observation in MM/DD/YYYY\n",
    "        - Province/State - Province or state of the observation (Could be empty when missing)\n",
    "        - Country/Region - Country of observation\n",
    "        - Last Update - Time in UTC at which the row is updated for the given province or country. (Not standardised and so please clean before using it)\n",
    "        - Confirmed - Cumulative number of confirmed cases till that date\n",
    "        - Deaths - Cumulative number of of deaths till that date\n",
    "        - Recovered - Cumulative number of recovered cases till that date\n",
    "\n",
    "    \n",
    "    :return dataframe \n",
    "    \"\"\"\n",
    "    df = pd.read_csv(path)\n",
    "    return df\n",
    "\n",
    "data = load_covid_dataset(path)\n",
    "\n",
    "\n",
    "# Do and log stuff\n",
    "profile = ProfileReport(df, title=\"Covid Dataset\")\n",
    "\n",
    "# end_run\n",
    "tracker.api.end_run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "profile"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pypads-padre-3.7",
   "language": "python",
   "name": "pypads-padre-3.7"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}